### PR TITLE
Rgxg: init at 0.1.2

### PR DESCRIPTION
**Pull request by hloeffler**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AndersonTorres** [comment]
@hloeffler can you rebase this, please?

**Comment by hloeffler** [comment]
@AndersonTorres 
Yes, sorry for the mess.

**Comment by AndersonTorres** [comment]
@hloeffler ping

**Comment by hloeffler** [comment]
@AndersonTorres rebased

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
@hloeffler change the code as shown below the changes below

File: pkgs/tools/text/rgxg/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  pname = "rgxg";
+  version = "0.1.2";
+
+  src = fetchurl {
```
[the remark this answers is not in the document]
Comment (hloeffler) [reply pkgs/tools/text/rgxg/default.nix]: Sure, thank you for pointing this out.

**Comment by hloeffler** [comment]
I think i need help,
if i build from repo, i do need the .git dir for ./version.sh

and now i'm down the rabbithole til:
https://github.com/NixOS/nixpkgs/issues/24169

**Comment by AndersonTorres** [comment]
`leaveDotGit=true; ` inside fetchFromGitHub.

Also, remember to interactively rebase it to only two commits.

File: pkgs/tools/text/rgxg/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (doronbehar) [reply pkgs/tools/text/rgxg/default.nix]: @AndersonTorres in the case of an autoconf build, moving to `fetchFromGitHub` means that `autoreconfHook` is needed as well, and specifically this project will require patching https://github.com/rgxg/rgxg/blob/master/version.sh . To get rid this hassle, I'd advise switching back to `fetchurl`.

**Comment by doronbehar** [comment]
> `leaveDotGit=true; ` inside fetchFromGitHub.

It's considered back practice from the point of reproducibility: https://github.com/NixOS/nixpkgs/issues/8567 .

**Comment by AndersonTorres** [comment]
I will try it locally now. It became very strange now...

File: pkgs/tools/text/rgxg/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (AndersonTorres) [reply pkgs/tools/text/rgxg/default.nix]: Oh, I see this now.

**Review by hloeffler — COMMENTED** [review]
done and rebased ( also added a comment)

File: pkgs/tools/text/rgxg/default.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  pname = "rgxg";
+  version = "0.1.2";
+
+  # fetchFromGitHub does not work because of how its build ( and leaveDotGit is considered bad )
```
Comment (doronbehar) [inline pkgs/tools/text/rgxg/default.nix:7]: LOL :). The situation is much more complicated then that so I think there's no need to explain. autoconf based tools are common in Nixpkgs and it's well known that they need an official "tarball" and can't easily be built with a bare git checkout, with or without the `.git` directory.

Comment (AndersonTorres) [reply pkgs/tools/text/rgxg/default.nix:7]: In this sense I would prefer `fetchzip`. At least it doesn't garble the hash sum.
But it doesn't matter now, it is working.

Comment (AndersonTorres) [inline pkgs/tools/text/rgxg/default.nix:7]:
```suggestion
```

File: pkgs/tools/text/rgxg/default.nix [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  src = fetchurl {
+    url = "https://github.com/rgxg/rgxg/releases/download/v${version}/rgxg-${version}.tar.gz";
```
Comment (AndersonTorres) [inline pkgs/tools/text/rgxg/default.nix:9]:
```suggestion
    url = "https://github.com/rgxg/rgxg/releases/download/v${version}/${pname}-${version}.tar.gz";
```

File: pkgs/tools/text/rgxg/default.nix (hunk 2 above)
Comment (hloeffler) [reply pkgs/tools/text/rgxg/default.nix]: :-)

**Comment by hloeffler** [comment]
I'm done :-)
Was easy.

(Thank you!)

File: pkgs/tools/text/rgxg/default.nix (hunk 2 above)
Comment (doronbehar) [reply pkgs/tools/text/rgxg/default.nix]: Why did we switch to `fetchzip`?

**Comment by AndersonTorres** [comment]
Now it is OK! Many thanks, guy! (or gal, I don't know :-) )

**Pull request merged**